foo_ui_columns/output_format.cpp

```diff
@@ -0,0 +1,63 @@
+#include "stdafx.h"
+
+#include "drop_down_list_toolbar.h"
+
+struct OutputFormatToolbarArgs {
+    using ID = uint32_t;
+    using ItemList = std::vector<std::tuple<ID, std::string>>;
+    static auto get_items()
+    {
+        static constexpr const auto BitDepths = {16, 24, 32};
+
+        static const auto Items = BitDepths
+            | ranges::views::transform([](ID bd) { return std::make_tuple(bd, std::to_string(bd) + "-bit"); })
+            | ranges::to<ItemList>();
+
+        if (!is_available())
+            return ItemList{};
+
+        return Items;
+    }
+    static ID get_active_item()
+    {
+        if (!is_available())
+            return {};
+
+        auto api = output_manager_v2::get();
+        outputCoreConfig_t config{};
+        api->getCoreConfig(config);
+        return config.m_bitDepth;
+    }
+    static void set_active_item(ID bitDepth)
+    {
+        if (!is_available())
+            return;
+
+        auto api = output_manager_v2::get();
+        outputCoreConfig_t config{};
+        api->getCoreConfig(config);
+        config.m_bitDepth = bitDepth;
+        api->setCoreConfig(config);
+    }
+    static void get_menu_items(uie::menu_hook_t& p_hook) {}
+    static void on_first_window_created()
+    {
+        if (!is_available())
+            return;
+
+        auto api = output_manager_v2::get();
+        callback_handle
+            = api->addCallback([] { DropDownListToolbar<OutputFormatToolbarArgs>::s_refresh_all_items_safe(); });
+    }
+    static void on_last_window_destroyed() { callback_handle.release(); }
+    static bool is_available() { return static_api_test_t<output_manager_v2>(); }
+    static service_ptr callback_handle;
+    static constexpr bool refresh_on_click = true;
```

Comment:
Not sure how this got there. Must have come from the `output_device.cpp` but judging by the fact that a static set of bit depths is a no-go, this could actually come in handy...